src/shared/test/prompts-test.js

```diff
@@ -1,56 +1,40 @@
-import { confirm, $imports } from '../prompts';
-
-function FakeConfirmDialog({
-  confirmAction,
-  message,
-  onCancel,
-  onConfirm,
-  title,
-}) {
-  return (
-    <div>
-      <h1>{title}</h1>
-      <p>{message}</p>
-      <button data-testid="confirm" onClick={onConfirm}>
-        {confirmAction}
-      </button>
-      <button data-testid="cancel" onClick={onCancel}>
-        Cancel
-      </button>
-    </div>
-  );
-}
+import { confirm } from '../prompts';
 
 describe('shared/prompts', () => {
   describe('confirm', () => {
     beforeEach(() => {
+      // This will cause the custom ConfirmModal to be used instead of
+      // window.confirm
       sinon.stub(window, 'confirm').returns(false);
-
-      $imports.$mock({
-        './components/ConfirmDialog': FakeConfirmDialog,
-      });
     });
 
     afterEach(() => {
       window.confirm.restore();
     });
 
-    function getCustomDialog() {
-      return document.querySelector('[data-testid="confirm-container"]');
+    function clickClose() {
+      const closeButton = getCustomDialog().querySelector(
+        '[aria-label="Close"]'
+      );
+      closeButton.click();
+    }
+
+    function clickCancel() {
+      const cancelButton = getCustomDialog().querySelector(
+        '[data-testid="cancel-button"]'
```

Comment:
We haven't decided yet whether `data-testid` attributes are part of the public "API" of a component. It might be convenient for public widgets to declare certain test IDs to be used in tests. I think this is fine for the moment, since the tests will break if the attribute ever changes.